### PR TITLE
Add Exception for empty rule

### LINKED ISSUE #3474: POST to /groups/{groupId}/artifacts/{artifactId}/rules with empty content results in NPE
## Description

**Registry
Version**: all
**Persistence type**: all

### Steps to Reproduce
1. Create a new artifact at some groupId + artifactId coordinates.
2. Send a POST request to `/groups/{groupId}/artifacts/{artifactId}/rules` to the coordinates in (1)
3. Ensure the `rule` section of the body of the request is empty

The server will respond with a `500` error (Null Pointer specifically).  The proper response is a `400` (Invalid Rule name).

### PR DESCRIPTION
**Pull request by Amoncy**
Hi @EricWittmann , 
I have made the changes suggested by you in my PR #3491. Please review the changes and let me know if this change is as expected. As mentioned by @carlesarnal I will add test cases for the same if the changes are correct.
Closes #3474

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java [hunk 1]
```diff
@@ -173,6 +173,15 @@ public List<RuleType> listGlobalRules() {
     @Audited(extractParameters = {"0", KEY_RULE})
     @Authorized(style=AuthorizedStyle.None, level=AuthorizedLevel.Admin)
     public void createGlobalRule(Rule data) {
+        String type = data.getType();
+        requireParameter("type", type);
```
Comment (carlesarnal) [inline app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:177]: Why is calling `requireParameter` not sufficient?

**Comment by carlesarnal** [comment]
Hi @Amoncy, I've added a small question. Apart from that, your code is not compiling. Before submitting a PR, I would encourage you to build the project locally (without tests) to confirm that at least the project is compiling.

**Comment by Amoncy** [comment]
Hi @carlesarnal,
I got a better understanding of the issue with the help of your comments.
I have made the changes. Can you please have a look?
Thank you~

File: app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java (hunk 1 above)
Comment (EricWittmann) [reply app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:183]: Because `requireParameter` doesn't check for empty string.

Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java:183]: That was my point, actually. The `requireParameter` only checks for nullability and, If we want to ensure that the type is not null, we should have it defined at the OpenAPI definition level as not null. Same goes for the config.

File: app/src/test/java/io/apicurio/registry/noprofile/rest/v2/GroupsResourceTest.java [hunk 2]
```diff
@@ -178,6 +178,53 @@ public void testDefaultGroup() throws Exception {
...
+    	//Test Rule type null
+    	Rule nullType = new Rule();
+    	nullType.setType(null);
+    	nullType.setConfig("TestConfig");
+    	given()
+	        	.when()
+	    			.pathParam("groupId", "testCreateArtifactRule")
+	    			.pathParam("artifactId", "testCreateArtifactRule/EmptyAPI/1")
+	    			.body(nullType)
+	    			.post("/registry/v2/groups/{groupId}/artifacts/{artifactId/rules}")
```
Comment (carlesarnal) [inline app/src/test/java/io/apicurio/registry/noprofile/rest/v2/GroupsResourceTest.java:197]: This test is failing due to a missing `}` at the end of artifactId.

**Review by carlesarnal — CHANGES_REQUESTED** [review]
@Amoncy I have figured out why your test is causing the application to hang, look at my comment.

File: app/src/test/java/io/apicurio/registry/noprofile/rest/v2/GroupsResourceTest.java (hunk 2 above)
Comment (Amoncy) [reply app/src/test/java/io/apicurio/registry/noprofile/rest/v2/GroupsResourceTest.java]: Thank you @carlesarnal for pointing that out. I have made the corrections now. Please review and let me know for any changes.

**Review by carlesarnal — APPROVED** [review]
This is looking good now, thanks @Amoncy!

**Comment by carlesarnal** [comment]
@Amoncy this branch cannot be safely rebased, do you mind doing it so we can merge this into main? Let me know if you need any assistance!

**Pull request merged**